ssl/statem/extensions_clnt.c

```diff
@@ -1389,6 +1439,44 @@ int tls_parse_stoc_maxfragmentlen(SSL_CONNECTION *s, PACKET *pkt,
     return 1;
 }
 
+int tls_parse_stoc_record_size_limit(SSL_CONNECTION *s, PACKET *pkt,
+                                  unsigned int context,
+                                  X509 *x, size_t chainidx) {
+    unsigned int peer_record_size_limit;
+
+    /*
+     * According to RFC 8449:
+     *
+     * A client MUST treat receipt of both "max_fragment_length" and
+     * "record_size_limit" as a fatal error, and it SHOULD generate
+     * an "illegal_parameter" alert.
+     */
+    if (USE_MAX_FRAGMENT_LENGTH_EXT(s->session)) {
+        SSLfatal(s, SSL_AD_ILLEGAL_PARAMETER, SSL_R_BAD_EXTENSION);
+        return 0;
+    }
+
+    if (PACKET_remaining(pkt) != 2 || !PACKET_get_net_2(pkt, &peer_record_size_limit)) {
```

Comment:
No need for the `PACKET_remaining` check here. `PACKET_get_net_2` will fail if there aren't 2 bytes available.